src/java.base/unix/classes/sun/nio/fs/UnixUserDefinedFileAttributeView.java

```diff
@@ -169,52 +169,50 @@ public int read(String name, ByteBuffer dst) throws IOException {
         assert (pos <= lim);
         int rem = (pos <= lim ? lim - pos : 0);
 
-        NativeBuffer nb;
-        long address;
-        if (dst instanceof sun.nio.ch.DirectBuffer) {
-            nb = null;
-            address = ((sun.nio.ch.DirectBuffer)dst).address() + pos;
+        if (dst instanceof sun.nio.ch.DirectBuffer buf) {
+            long address = buf.address() + pos;
+            int n = read(name, address, rem);
```

Comment:
We need to add reachabilityFence here, don't mind if we do it now while changing this code or later.
That is, it should be try { n = read(name, address, rem); } finally { Reference.reachabilityFence(buf); }

Reply:
Never heard of this before, just read the docs the first time. Do I get this right, that this helps to protect against instant finalization of the buffer in e.g. `attrView.read("foo", ByteBuffer.allocateDirect(42))`?

What about the `write` case?

Reply:
Yes, everywhere where the direct memory is accessed (so read and write here). I don't mind if we do it as a separate issue or here, I only suggested it here because it's changed the new code in this patch.